### PR TITLE
feat: http operation extensions

### PR DESCRIPTION
**Pull request by mallachari**
Addresses: https://github.com/stoplightio/platform-internal/issues/7772
~Requires: https://github.com/stoplightio/types/pull/97~

Transforming operation `x-extensions` properties to `HttpOperation` `extensions` property.

### CODE REVIEW

File: src/oas/accessors.ts [hunk 1]
```diff
@@ -25,3 +26,15 @@ const isValidOasParameter = (parameter: Partial<ParamTypeBase>): parameter is Pa
 export function getOasTags(tags: unknown): string[] {
   return Array.isArray(tags) ? tags.filter(tag => typeof tag !== 'object').map(String) : [];
 }
+
+export function getExtensions(target: object): Extensions {
```
Comment (P0lip) [inline src/oas/accessors.ts:30]:
```suggestion
export function getExtensions(target: Record<string, unknown>): Extensions {
```

File: src/oas/accessors.ts [hunk 2]
```diff
@@ -25,3 +26,15 @@ const isValidOasParameter = (parameter: Partial<ParamTypeBase>): parameter is Pa
...
+  return Object.keys(target)
+    .filter(key => key.startsWith('x-'))
+    .reduce(
+      (obj, key) => ({
+        ...obj,
+        [key]: target[key],
+      }),
+      {},
+    );
+}
```
Comment (P0lip) [inline src/oas/accessors.ts:40]:
```suggestion
  return Object.fromEntries(Object.entries(target).filter(([key]) => key.startsWith('x-'));
}
```

Comment (P0lip) [reply src/oas/accessors.ts:40]: I think we may want to exclude extensions we already consume, i.e. `x-internal`. Currently, it's added twice, once as `-internal` and the other time within `extensions`

Comment (mallachari) [reply src/oas/accessors.ts:40]: Yeah, I was aware of `x-internal` and thought we might still want to keep it in the object to not have exceptions. But can surely exclude it.

**Comment by mallachari** [comment]
@stoplightio/plaid could you take a look? We need your approval here 😉

File: src/oas/accessors.ts [hunk 3]
```diff
@@ -25,3 +28,7 @@ const isValidOasParameter = (parameter: Partial<ParamTypeBase>): parameter is Pa
 export function getOasTags(tags: unknown): string[] {
   return Array.isArray(tags) ? tags.filter(tag => typeof tag !== 'object').map(String) : [];
 }
+
+export function getExtensions(target: Record<string, unknown>): Extensions {
+  return fromPairs(Object.entries(target).filter(([key]) => key.startsWith('x-') && !ROOT_EXTENSIONS.includes(key)));
```
Comment (Nezteb) [inline src/oas/accessors.ts:33]: Why are we excluding `x-internal`?

Comment (mallachari) [reply src/oas/accessors.ts:33]: It is already being included separately as `internal` property in `HttpOperation`, so we don't want to double it

**Review by marbemac — APPROVED** [review]

**Review by P0lip — APPROVED** [review]

**Pull request merged**